libindy/src/services/metrics/mod.rs

```diff
@@ -127,10 +118,10 @@ mod test {
     fn test_counters_are_initialized_as_zeros() {
         let metrics_service = MetricsService::new();
         for index in (0..MetricsService::commands_count()).rev() {
-            assert_eq!(metrics_service.queued_commands_count.borrow()[index as usize], 0);
-            assert_eq!(metrics_service.queued_commands_duration_ms.borrow()[index as usize], 0);
-            assert_eq!(metrics_service.executed_commands_count.borrow()[index as usize], 0);
-            assert_eq!(metrics_service.executed_commands_duration_ms.borrow()[index as usize], 0);
+            assert_eq!(metrics_service.queued_counters.borrow()[index as usize].count, 0);
+            assert_eq!(metrics_service.queued_counters.borrow()[index as usize].duration_ms_sum, 0);
+            assert_eq!(metrics_service.executed_counters.borrow()[index as usize].count, 0);
+            assert_eq!(metrics_service.executed_counters.borrow()[index as usize].duration_ms_sum, 0);
```

Comment:
I would suggest to create a unit test suite for MetricCounter class, so that you don't need to do all those repetitions here. And, btw, in your test you're missing check that buckets are also initialized as zeros.